.github/workflows/beam_PreCommit_Go.yml

```diff
@@ -1,37 +1,69 @@
-name: beam_PreCommit_Go
+name: Go Precommit
 
 on:
   push:
     tags: ['v*']
     branches: ['master', 'release-*']
-  pull_request:
-    branches: ['master']
+  pull_request_target:
+    branches: ['master', 'release-*' ]
     paths: ['model/**', 'sdks/go.**', 'release/**']
   issue_comment:
     types: [created]
   schedule:
     - cron: '* */6 * * *'
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: read
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
 
 jobs:
   beam_PreCommit_Go:
-        if: ${{github.event.issue.pull_request}} || ${{github.event.comment.body == 'Run Go PreCommit'}} || ${{github.event.schedule}}
-        runs-on: [self-hosted, ubuntu-20.04]
-        name: beam_PreCommit_Go
-        steps:       
-          - name: Git checkout
-            uses: actions/checkout@v3
-          - name: Install Java
-            uses: actions/setup-java@v3.8.0
-            with:
-              distribution: 'zulu'
-              java-version: '8'
-          - name: Install Go
-            uses: actions/setup-go@v4
-            with:
-              go-version: '1.20'
-          - name: Setup Gradle
-            uses: gradle/gradle-build-action@v2
-            with:
-              cache-read-only: false
-          - name: run goPreCommit script
-            run: ./gradlew :goPreCommit
\ No newline at end of file
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Go PreCommit'
+    runs-on: [self-hosted, ubuntu-20.04]
+    name: beam_PreCommit_Go
+    steps:       
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install GH CLI
+        uses: dev-hanz-ops/install-gh-cli-action@v0.1.0
```

Comment:
Do we need this for the rerun-job-action? If so, should it just be part of the action itself instead of each workflow doing it? Or, maybe it should just be part of the image? I'd lean towards that since its going to be common to pretty much every workflow.

Also, we'll need infra approval to use this action unfortunately if we decide to use it.

Reply:
Plan is to have cli installed in image in the future. While most calls to api are made with curl we need GH CLI to wait on workflow cancellation. 

For infra approval im not sure for which part? That we use external action to install? If yes i can change to script and install the CLI that way

Reply:
If the issue is the custom action we can put a simple and trasparent script action for the time being
```      - name: Install GH Cli
        run: |
          wget https://github.com/cli/cli/releases/download/v2.31.0/gh_2.31.0_linux_amd64.tar.gz
          tar -xvf gh_2.31.0_linux_amd64.tar.gz
          sudo mv gh_2.31.0_linux_amd64/bin/gh /usr/local/bin
```
          
          
In the meantime ill work on getting CLI into the image and shipping that image

Reply:
A script might be simpler vs asking infra to approve the action.

I do still think it makes sense to include this as part of `rerun-job-action` instead of this workflow since each workflow that uses `rerun-job-action` will need this though

Reply:
Agreed. Moved to `rerun-job-action`.